### ISSUE TITLE
Expose more rule error output in CI

### ISSUE THREAD

**Issue by ArmaanMcleod** [microsoft/ps-rule#140]
**Is your feature request related to a problem? Please describe.**

Continuing from https://github.com/microsoft/PSRule/discussions/962

As shown here:

![Screenshot 2022-02-06 215200](https://user-images.githubusercontent.com/20082136/153095314-823a9ad5-85b3-43c9-b79c-1cbe4d9fffe8.jpg)

Sometimes it would be useful to show more error output here to understand what the issue is. 

**Describe the solution you'd like**

 Catch the specific exception type in https://github.com/microsoft/ps-rule/blob/main/powershell.ps1#L181. 

**Describe alternatives you've considered**

Other solutions:

- Add all exception messages with `$_.Exception.Message` inside the error message: https://github.com/microsoft/ps-rule/blob/main/powershell.ps1#L182. May not want to do this because some messages should not be shown in CI, such as the `Pipeline Stopped...` exception.
- Add a CI option to expose full exception message. Could be useful if customer wants to set this option in `ps-rule.yaml` to control the verbosity of the output. 

**Additional context**

Should be included for Azure Pipelines as well.

**Comment by BernieWhite** [comment]
@ArmaanMcleod Ok let's start with this. But let's double check that if an exception is raised within a rule it does not terminate the pipeline.

```powershell
Rule 'SomeRule' {
   throw 'Test'
}
```

It should be reported as a rule error but allow other rules to continue processing.

**Comment by ArmaanMcleod** [comment]
@BernieWhite Great. From what I've tested it doesn't terminate the pipeline if a rule throws an exception.
